src/operator/fusion/fused_op-inl.h

```diff
@@ -550,7 +550,10 @@ __device__ inline DType sigmoid(const DType val) {
 
 template <typename DType>
 __device__ inline DType softrelu(const DType val) {
-  return logf(1 + expf(val));
+  // Avoid overflow of exp for large inputs.
+  // The threshold 20 is chosen such that softrelu(a) = a
+  // for a > 20 using floating precision.
+  return val > 20 ? val : logf(1 + expf(val));
```

Comment:
There's reference implementation in mkldnn that uses a different threshold and also does type cast. https://github.com/intel/mkl-dnn/commit/23f45a2e4cb4158e6111d3e59d91b66038636edb#diff-b476290dff3dd01b30438be4ae713928R174

Shall we adopt the same logic for consistency?

Reply:
From the test I did - difference between `logf(1 + expf(20)) - 20` is 0 already, so I think there should not be any difference (and if anything, going to max float should give less precision).

Reply:
I took the reference threshold from MxNets own [mshadow impl](https://github.com/apache/incubator-mxnet/blob/ea2dabaae5e1453572c9105f77322977bd14c108/src/operator/mshadow_op.h#L420).

`log(1 + exp(v))` is litteraly the same floating point value as `v` for float32 and `14.2109584 <= v < 88.72284`. For `v < 14.2109584`, there is a non-zero difference `log(1 + exp(v)) - v`, and for `v >= 88.72284`, `exp(v)` is `+inf`.

Given that choosing any value in that range should yield exactly the same results, I would intuitively pick a lower value since it skips the `log(1+exp(v))` computation in favour of identity `v` earlier.

Although, I guess, 88.72284 is slightly less "magic number"  than 20, since it's `log(FLT_MAX)`.

I am not particularly committed to either version, so the final decision is up to you

Reply:
OK